src/channel_server/player_data_provider.cpp

```diff
@@ -235,7 +235,9 @@ auto player_data_provider::stop_following(ref_ptr<player> follower) -> void {
 }
 
 auto player_data_provider::disconnect() -> void {
-	for (auto &kvp : m_players) {
+	auto m_players_copy = hash_map<game_player_id, ref_ptr<player>>(m_players);
```

Comment:
C++ uses value type semantics by default. Unless `m_players` is a pointer, it will make a full copy of the value itself.